src/registrar/models/domain.py

```diff
@@ -1582,45 +1616,21 @@ def _delete_hosts_if_not_used(self, hostsToDelete: list[str]):
     def _fetch_cache(self, fetch_hosts=False, fetch_contacts=False):
         """Contact registry for info about a domain."""
         try:
-            # get info from registry
             data_response = self._get_or_create_domain()
             cache = self._extract_data_from_response(data_response)
-
-            # remove null properties (to distinguish between "a value of None" and null)
-            cleaned = self._remove_null_properties(cache)
-
-            if "statuses" in cleaned:
-                cleaned["statuses"] = [status.state for status in cleaned["statuses"]]
-
-            cleaned["dnssecdata"] = self._get_dnssec_data(data_response.extensions)
-
-            # Capture and store old hosts and contacts from cache if they exist
-            old_cache_hosts = self._cache.get("hosts")
-            old_cache_contacts = self._cache.get("contacts")
-
-            if fetch_contacts:
-                cleaned["contacts"] = self._get_contacts(cleaned.get("_contacts", []))
-                if old_cache_hosts is not None:
-                    logger.debug("resetting cleaned['hosts'] to old_cache_hosts")
-                    cleaned["hosts"] = old_cache_hosts
-
-            if fetch_hosts:
-                cleaned["hosts"] = self._get_hosts(cleaned.get("_hosts", []))
-                if old_cache_contacts is not None:
-                    cleaned["contacts"] = old_cache_contacts
-
-            # if expiration date from registry does not match what is in db,
-            # update the db
-            if "ex_date" in cleaned and cleaned["ex_date"] != self.expiration_date:
-                self.expiration_date = cleaned["ex_date"]
-                self.save()
+            cleaned = self._clean_cache(cache, data_response)
+            self._update_hosts_and_contacts(cleaned, fetch_hosts, fetch_contacts)
```

Comment:
(Blocking) This never returns the value of cleaned, so the functions below this are using an outdated version of the cache